lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_upload_pipeline_action.rb

```diff
@@ -0,0 +1,70 @@
+module Fastlane
+  module Actions
+    class BuildkiteUploadPipelineAction < Action
+      def self.run(params)
+        pipeline_env_file = params[:pipeline_env_file]
+        pipeline_file = params[:pipeline_file]
+        branch = params[:branch]
+        commit = params[:commit]
+
+        UI.user_error!("Pipeline file not found: #{pipeline_file}") unless File.exist?(pipeline_file)
+
+        UI.message "Uploading pipeline #{pipeline_file} on branch #{branch}, commit #{commit}"
+
+        ENV['BUILDKITE_BRANCH'] = branch
+        ENV['BUILDKITE_COMMIT'] = commit
+
+        if pipeline_env_file && File.exist?(pipeline_env_file)
+          UI.message("Sourcing environment file: #{pipeline_env_file}")
+
+          sh(". #{pipeline_env_file} && buildkite-agent pipeline upload #{pipeline_file}")
+        else
+          sh('buildkite-agent', 'pipeline', 'upload', pipeline_file)
+        end
+      end
+
+      def self.description
+        # https://buildkite.com/docs/agent/v3/cli-pipeline#uploading-pipelines
+        'Uploads a pipeline to Buildkite, adding all its steps to the current build'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :pipeline_env_file,
+            description: 'The path to the environment variables file to be sourced before uploading the pipeline',
+            optional: true,
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :pipeline_file,
+            description: 'The path to the Buildkite pipeline file',
```

Comment:
Description updated on 07661f76ab61bc32a665e8837c27c68a2a02c5f3. I've also changed the order of the `ConfigItem`s so that the required parameter is the first one and the optional ones come later.